ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_facts/uplinks.py

```diff
@@ -181,150 +181,232 @@ def uplinks(self: EosDesignsFacts) -> list:
-                    # Invalid length of input variables. Skipping
-                    continue
+    def _get_p2p_vrfs_uplink(self: EosDesignsFacts, uplink_index: int, uplink_interface: str, uplink_switch: str, uplink_switch_interface: str) -> dict:
+        """
+        Return a single uplink dictionnary for uplink_type p2p-vrfs
+        """
+        uplink_switch_facts: EosDesignsFacts = self.shared_utils.get_peer_facts(uplink_switch, required=True)
+        uplink = {
+            "interface": uplink_interface,
+            "peer": uplink_switch,
+            "peer_interface": uplink_switch_interface,
+            "peer_type": uplink_switch_facts.type,
+            "peer_is_deployed": uplink_switch_facts.is_deployed,
+            "peer_bgp_as": uplink_switch_facts.bgp_as,
+            "type": "p2p_vrfs",
+        }
+        if self.shared_utils.uplink_interface_speed is not None:
+            uplink["speed"] = self.shared_utils.uplink_interface_speed
+
+        if self.shared_utils.uplink_bfd:
+            uplink["bfd"] = True
+
+        if self.shared_utils.uplink_switch_interface_speed is not None:
+            uplink["peer_speed"] = self.shared_utils.uplink_switch_interface_speed
+
+        # if self.shared_utils.uplink_ptp is not None:
+        #    uplink["ptp"] = self.shared_utils.uplink_ptp
+        # elif self.shared_utils.ptp_enabled:
+        #    uplink["ptp"] = {"enable": True}
+
+        # if self.shared_utils.uplink_macsec is not None:
+        #    uplink["mac_security"] = self.shared_utils.uplink_macsec
+
+        # if self.shared_utils.underlay_multicast is True and uplink_switch_facts.shared_utils.underlay_multicast is True:
+        #    uplink["underlay_multicast"] = True
+
+        # if self.shared_utils.link_tracking_groups is not None:
+        #    uplink["link_tracking_groups"] = []
+        #    for lt_group in self.shared_utils.link_tracking_groups:
+        #        uplink["link_tracking_groups"].append({"name": lt_group["name"], "direction": "upstream"})
+
+        uplink["subinterfaces"] = self._uplink_sub_interfaces(
+            uplink_interface, uplink_switch, uplink_switch_interface, uplink_switch_facts, uplink_index, self.shared_utils.uplink_structured_config
+        )
 
-                uplink_switch = uplink_switches[uplink_index]
-                if uplink_switch is None or uplink_switch not in self.shared_utils.all_fabric_devices:
-                    # Invalid uplink_switch. Skipping.
+        return uplink
+
+    def _uplink_sub_interfaces(
+        self: EosDesignsFacts,
+        uplink_interface: str,
+        uplink_switch: str,
+        uplink_switch_interface: str,
+        uplink_switch_facts: EosDesignsFacts,
+        uplink_index: int,
+        struct_cfg: dict | None,
+    ) -> list:
+        subinterfaces = []
+        for tenant in self.shared_utils.filtered_tenants:
+            # TODO this could lead to duplicates maybe - maybe use VRF names
+            for vrf in tenant["vrfs"]:
```

Comment:
Handle VRF default. Maybe my reusing existing p2p function.
Also inject VRF default into filtered tenants. (id 1)
Also a knob for `include_vrf_default_on_uplink`